Revise motivation (#17)
SaranjeetKaur/isc_proposal_translations: README.md

[![build-status](https://github.com/SaranjeetKaur/isc_proposal_translations/actions/workflows/publish-proposal.yaml/badge.svg)](https://github.com/SaranjeetKaur/isc_proposal_translations/actions/workflows/publish-proposal.yaml)

> This repository contains the proposal for "Consistent Translations Glossary across R". The proposal aims to improve the technical infrastructure of the translations process across R projects. There are many volunteer translation efforts across the R community (messages in base R/recommended packages, rOpenSci materials such as their package development guide, Bioconductor training materials, R books such as R for Data Science, etc). These projects often have their own glossaries, since the benefits of using a glossary in translation projects are widely recognised (improves translation quality, reduces translation effort). However, developing and maintaining separate glossaries for each translation project reduces these benefits - different projects may use different translations for certain terms, effort to identify relevant terms and translate them is duplicated across projects putting unnecessary burden on volunteers. Some projects do not have the volunteer capacity to translate the glossary into multiple languages, or may not even use a glossary. Currently it is difficult for projects to take full advantage of existing glossaries due to diversity in the choice of translation infrastructure. These issues will become even more important if internationalization of R help pages becomes more common (ref R Consortium-funded project: [tooling-for-internationalization-of-r-help-pages](https://r-consortium.org/all-projects/2023-group-2.html#tooling-for-internationalization-of-r-help-pages))

Reviewer: > messages in base R/recommended packages, ...

'individual packages' is another highly-decentralized source of translations, may be worth including. I know {data.table} would benefit from such a central effort.

Separately, I know the well-outdated Translations page here:

https://developer.r-project.org/Translations30.html

Points to the "ISI glossary of statistical terms" (which link is now dead, RIP 🪦). This paragraph makes me appreciate the benefit of centralized resources for the R universe, but also wonder why stop at that boundary? I.e., why only aim to improve translation for R developers, when this might benefit e.g. the Python community as well?

(just something to consider)

> An example of diverging translation of glossary terms is provided by the rOpensci English-Spanish glossary:

Reviewer: ```suggestion
An example of diverging translation of glossary terms is found in the rOpensci English-Spanish glossary:
```

| English | Spanish (rOpenSci) | Spanish (R project)                        |
|--------|--------------------|--------------------------------------------|

> | path    | ruta de acceso     | ruta (file path) / trayecto (polygon path) |   

Reviewer 2: Actually, the `polygon` part is just translating `path` into Spanish in the wrong context, I would say... But it s a very good example.

> that could be due to the term arising in different contexts though. There are many examples of diverging translations that might have been avoided if a glossary had been used, e.g. in base R "loop" is variously translated as "bucle", "loop", "ciclo" - both rOpensci and R project glossaries agree on "bucle" being the preferred term; sometimes "camino" is used rather than "ruta" for a file path. These don't change the meaning but may read a bit strangely. Since, there are limited glossaries to compare, it is difficult to provide more examples.

Reviewer 2: This is actually a tricky one. What Spanish are we talking about? Spanish from Spain, or Chile, or Argentina, etc. are all notably different in what words are the "right" ones. So the problem is actually pretty complicated. I can easily see that is an R package is developed in, Chile, they will use some words in Spanish that make sense to them, but not to someone from Spain - and the other way around - despite both being perfectly fine translations.

Reviewer 2: And, obviously, the same might apply to other major languages, spoken in multiple countries.

Reviewer: Technically, localized Spanish should take advantage of the extended locale codes (I forget the technical term), I.e. `es_ES`, `es_CL`, and `es_AR`, resp., like we have `en_US` vs. `en_GB` already for English (mostly about spelling). See e.g. https://www.gnu.org/software/gettext/manual/gettext.html#Specifying-a-Priority-List-of-Languages
